src/mame/act/v9kdmaib.cpp

```diff
@@ -0,0 +1,292 @@
+// license:BSD-3-Clause
+
+#include "emu.h"
+#include "v9kdmaib.h"
+
+#include "logmacro.h"
+
+/*
+ * Emulation of the Victor 9000 DMA Interface Board, which interfaces to the
+ * Xebec S1410 Hard Disk controller.
+ *
+ * Source: https://bitsavers.org/pdf/victor/victor9000/Victor_9000_Sirius_1_Hard_Disk_Subsystem.pdf
+ *
+ * Compared to the typical SCSI/SASI controller device, the DMA Interface Board
+ * acts more like a passthrough, and the usual state machine is primarily
+ * implemented on the main CPU.
+ */
+
+DEFINE_DEVICE_TYPE(V9KDMAIB, v9kdmaib_device, "v9kdmaib", "Victor 9000 DMA Interface Board")
+
+v9kdmaib_device::v9kdmaib_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	v9kdmaib_device(mconfig, V9KDMAIB, tag, owner, clock)
+{
+}
+
+v9kdmaib_device::v9kdmaib_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock) :
+	nscsi_device(mconfig, type, tag, owner, clock),
+	nscsi_slot_card_interface(mconfig, *this, DEVICE_SELF),
+	m_dma_r(*this, 0xff),
+	m_dma_w(*this),
+	m_irq_handler(*this),
+	m_dma_on(false),
+	m_dma_write(false),
+	m_dma_addr(0),
+	m_asserting_ack(false),
+	m_non_dma_req(false),
+	m_ctrl(0),
+	m_data(0),
+	m_bus_ctrl(0),
+	m_irq_state(false)
+{
+}
+
+uint8_t v9kdmaib_device::read(offs_t offset)
+{
+	// Lower address bits are ignored
+	if (offset >= 0x80)
+	{
+		offset &= ~0x1f;
+	}
+	else
+	{
+		offset &= ~0xf;
+	}
+
+	uint8_t data = 0xff;
+
+	switch (offset) {
+		case R_CONTROL:
+			// Write only
+			break;
+		case R_DATA:
+			if (m_non_dma_req)
+			{
+				m_non_dma_req = false;
+				scsi_bus->data_w(scsi_refid, 0);
+				m_data = scsi_bus->data_r();
+				m_asserting_ack = true;
+				scsi_bus->ctrl_w(scsi_refid, S_ACK, S_ACK);
```

Comment:
You need to check `machine().side_effects_disabled()` before doing stuff that affects state on reads.  If you don’t, it causes chaos when debugger memory windows are open.